src/Signature/SignatureV4.php

```diff
@@ -43,9 +50,13 @@ public function signRequest(
         if ($token = $credentials->getSecurityToken()) {
             $parsed['headers']['X-Amz-Security-Token'] = [$token];
         }
-
         $cs = $this->createScope($sdt, $this->region, $this->service);
         $payload = $this->getPayload($request);
+
+        if ($this->unsigned) {
```

Comment:
This works, yet the check here feels a bit weird, since `getPayload` has already checked both `$this->unsigned` and `http` scheme, could we check whether `$payload` is `self::UNSIGNED_PAYLOAD` instead?